# Answering Questions 7-9:

In [ ]:
# Load necessary libraries
library(xts)
library(quantmod)
library(PerformanceAnalytics)
library(dplyr)
library(ggplot2)

In [3]:
# Load the data
sp500_data_raw <- read.csv("sp500_2023_2024.csv")

# Remove rows with NA value in prcod
sp500_noNA <- sp500_data_raw %>%
    filter(!is.na(prcod))

# Convert datadate to Date format
sp500_noNA$datadate <- as.Date(sp500_noNA$datadate, format = "%d/%m/%Y")
sp500_data <- sp500_noNA

head(sp500_data)


,tic,datadate,conm,exchg,sic,cshtrd,prccd,prchd,prcld,prcod,gvkey
,<chr>,<date>,<chr>,<int>,<int>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<int>
1,PNW,2023-01-03,PINNACLE WEST CAPITAL CORP,11,4911,1442534,74.63,76.4125,73.380,76.25,1075
2,PNW,2023-01-04,PINNACLE WEST CAPITAL CORP,11,4911,954218,75.39,76.0950,74.630,75.10,1075
3,PNW,2023-01-05,PINNACLE WEST CAPITAL CORP,11,4911,994775,73.65,75.0950,73.305,74.88,1075
4,PNW,2023-01-06,PINNACLE WEST CAPITAL CORP,11,4911,729808,75.46,76.0200,74.480,74.49,1075
5,PNW,2023-01-09,PINNACLE WEST CAPITAL CORP,11,4911,656127,75.55,76.4800,75.240,75.24,1075
6,PNW,2023-01-10,PINNACLE WEST CAPITAL CORP,11,4911,763254,75.65,75.6950,74.880,75.31,1075


## Question 7:
### Which ticker has the largest positive mean return?

In [ ]:
# Add daily return column next to prccd

sp500_data <- sp500_data %>%
    arrange(tic, datadate) %>%
    group_by(tic) %>%
    mutate(daily_return = (prccd / lag(prccd)) - 1) %>%
    ungroup() %>%
    relocate(daily_return, .after = prccd)

#head(sp500_data)

# Find the mean daily return for each stock

sp500_mean_returns <- sp500_data %>%
    group_by(tic) %>%
    summarize(mean_daily_return = mean(daily_return, na.rm = TRUE)) %>%
    arrange(desc(mean_daily_return))

head(sp500_mean_returns, 10)

tic,mean_daily_return
<chr>,<dbl>
PLTR,0.005785119
COIN,0.005497913
GEV,0.005246567
VST,0.004044824
META,0.003367155
RCL,0.003353639
TSLA,0.003297285
NVDA,0.003160234
AXON,0.002834236


### Findings:
PLTR has the highest mean daily return at 0.58%.

## Question 8:
### Which company has the largest positive mean return?

In [6]:
# Add daily return column next to prccd

sp500_data <- sp500_data %>%
    arrange(conm, datadate) %>%
    group_by(conm) %>%
    mutate(daily_return = (prccd / lag(prccd)) - 1) %>%
    ungroup() %>%
    relocate(daily_return, .after = prccd)

#head(sp500_data)

# Find the mean daily return for each stock

sp500_mean_returns <- sp500_data %>%
    group_by(conm) %>%
    summarize(mean_daily_return = mean(daily_return, na.rm = TRUE)) %>%
    arrange(desc(mean_daily_return))

head(sp500_mean_returns, 10)

conm,mean_daily_return
<chr>,<dbl>
PALANTIR TECHNOLOG INC,0.005785119
COINBASE GLOBAL INC,0.005497913
GE VERNOVA INC,0.005246567
VISTRA CORP,0.004044824
FOX CORP,0.003403895
META PLATFORMS INC,0.003367155
ROYAL CARIBBEAN GROUP,0.003353639
TESLA INC,0.003297285
NVIDIA CORP,0.003160234


Palantir Technologies Inc has the highest mean daily return among the S&P 500 stocks in the dataset

## Question 9:
### Which industry is represented by the most companies?

In [8]:
sic_companies <- sp500_data %>%
    group_by(sic) %>%
    summarize(company_count = n_distinct(conm)) %>%
    arrange(desc(company_count)) %>%
    ungroup()

head(sic_companies, 10)

sic,company_count
<int>,<int>
6798,28
7370,23
3674,15
6020,14
4911,13
4931,13
7372,12
6282,11
6331,11


### Findings:
The industry with SIC code 6798 (Real Estate Investment Trusts) is represented by the most companies, with a total of 28 companies in the S&P 500 between 2023 to 2024.